src/decision_process.md

```diff
@@ -0,0 +1,51 @@
+This section documents the work-in-progress Rust language team decision
+process. This process, and the `rustbot` tooling to support it, does not yet
+have a finished implementation. This document serves to explain the intended
+process, for the purposes of ongoing implementation.
+
+## Prioritized principles of Rust team consensus decision-making
+
+These are in order of priority. They're intended to be general enough that they
+could apply to any Rust governance team, not just the language team.
+
+- **Treasure dissent.** When someone raises a concern, that's a chance to
+  improve the design, and to discover and explore underlying values. Dissent
+  should be an amicable, cooperative process.
+- **Understand and cooperatively resolve concerns.** We cannot resolve a
+  concern without first understanding it, including the underlying values
+  motivating it. We should demonstrate that understanding by documenting the
+  concern. We should consider the tradeoffs and the impacts on users, through
+  the Rust design principles. We should seek out and favor satisfying solutions
+  (those that satisfy everyone's values) over
+  [satisficing](https://en.wikipedia.org/wiki/Satisficing) solutions (those
+  that are just good enough for people to accept them as a compromise among
+  conflicting values, without actually being happy with the outcome).
+- **Don't force an irreversible decision.** We should make as many decisions
+  reversible as possible. Even when a decision seems irreversible (e.g.,
+  stabilization), we should strive to find a subset that still allows
+  addressing the use case, or a subset that will support evaluation and
+  data-gathering to enable making the decision in the future. If that's not
```

Comment:
```suggestion
  reversible as possible. When a decision is irreversible (e.g.,
  stabilization of a feature), if we are not certain which alternative is best, then we should strive to find an intermediate option that we are confident in and allows
  addressing the use case, or one that will support evaluation and
  data-gathering to enable making a more informed decision in the future. If that's not
```

Reply:
I'm not actually confident in my own suggestion here. All I know is that I find the current phrasing of this bullet hard to understand.

Namely, I cannot tell what it is actually suggesting we do in general.

The way it was originally written, with the chosen example of "stabilization", it sounds like we should be avoiding stabilizing *anything* new.

I tried to qualify that in my suggested text by saying there are obvious cases where the team is going to move forward, and that this is more relevant to the situations where we are not sure, or have disagreements/concerns.

(I also tried to remove the use of the word "subset", because the text as written doesn't have a clear universe or superset that you're taking a subset of. The *example* of stabilizing a feature or set of features *does* allow for a subset, but that's just an example, not the big picture idea...)

Reply:
> The way it was originally written, with the chosen example of "stabilization", it sounds like we should be avoiding stabilizing anything new.

The heading here is "Don't force an irreversible decision". "force" is the operative word; we can absolutely make irreversible (or very-hard-to-reverse) decisions, such as stabilization, but we should be *incredibly* hesitant forcing such a decision over objections.

I'll rewrite this to try to make that more clear.